cirq/sim/clifford_simulator_test.py

```diff
@@ -0,0 +1,359 @@
+    state.apply_unitary(cirq.H(q0))
+    state.apply_unitary(cirq.S(q0))
+
+    assert(state.tableau._str_full_() == 
+        "stable | destable\n"
+        "-------+----------\n"
+        "+ Y0   | + Z0  \n"  
+        "+   Z1 | +   X1\n") 
+
+def test_ch_form_H():
+    (q0,q1) = (cirq.LineQubit(0),cirq.LineQubit(1))
+    state = cirq.CliffordState(qubit_map={q0: 0,q1: 1})
+    with pytest.raises(ValueError, match="|y> is equal to |z>"):
+        state.CH_form._H_decompose(0,1,1,0)
+
+def test_clifford_ch_form_repr_full():
+    (q0,q1) = (cirq.LineQubit(0),cirq.LineQubit(1))
+    state = cirq.CliffordState(qubit_map={q0: 0,q1: 1})
+
+    assert(state.CH_form._repr_full() == 
+        "omega: 1.00\n"
+        "G:\n"
+        "[[1 0]\n"
+        " [0 1]]\n"
+        "F:\n"
+        "[[1 0]\n"
+        " [0 1]]\n"
+        "M:\n"
+        "[[0 0]\n"
+        " [0 0]]\n"
+        "gamma: [0 0]\n"
+        "v: [0 0]\n"
+        "s: [0 0]\n") 
+
+def test_clifford_circuit():
+    (q0,q1) = (cirq.LineQubit(0),cirq.LineQubit(1))
+    circuit = cirq.Circuit()
+
+    np.random.seed(0)
+
+    for _ in range(100):
+        x = np.random.randint(7)
+
+        if x == 0:
+            circuit.append(cirq.X(np.random.choice((q0,q1))))
+        elif x == 1:
+            circuit.append(cirq.Z(np.random.choice((q0,q1))))
+        elif x == 2:
+            circuit.append(cirq.Y(np.random.choice((q0,q1))))
+        elif x == 3:
+            circuit.append(cirq.S(np.random.choice((q0,q1))))
+        elif x == 4:
+            circuit.append(cirq.H(np.random.choice((q0,q1))))
+        elif x == 5:
+            circuit.append(cirq.CNOT(q0,q1))
+        elif x == 6:
+            circuit.append(cirq.CZ(q0,q1))
+
+    clifford_simulator = cirq.CliffordSimulator()
+    wave_function_simulator = cirq.Simulator()
+
+    np.testing.assert_almost_equal(clifford_simulator.simulate(circuit).final_state.wave_function(),
+        wave_function_simulator.simulate(circuit).final_state)
+
+def test_clifford_circuit_2():
+    (q0,q1) = (cirq.LineQubit(0),cirq.LineQubit(1))
+    circuit = cirq.Circuit()
+
+    np.random.seed(2)
+
+    for _ in range(10):
+        x = np.random.randint(7)
+
+        if x == 0:
+            circuit.append(cirq.X(np.random.choice((q0,q1))))
+        elif x == 1:
+            circuit.append(cirq.Z(np.random.choice((q0,q1))))
+        elif x == 2:
+            circuit.append(cirq.Y(np.random.choice((q0,q1))))
+        elif x == 3:
+            circuit.append(cirq.S(np.random.choice((q0,q1))))
+        elif x == 4:
+            circuit.append(cirq.H(np.random.choice((q0,q1))))
+        elif x == 5:
+            circuit.append(cirq.CNOT(q0,q1))
+        elif x == 6:
+            circuit.append(cirq.CZ(q0,q1))
+
+    circuit.append(cirq.measure(q0))
+    result = cirq.CliffordSimulator().run(circuit, repetitions=100)
+
+    assert sum(result.measurements['0'])[0] < 80
+    assert sum(result.measurements['0'])[0] > 20
+
+def test_clifford_circuit_3():
```

Comment:
Is this just a copy paste of the previous test? Use `pytest.mark.parameterize` t omerge them.

Reply:
Done.